src/components/buttons/hold-to-authorize/UnlockingSpinner.js

```diff
@@ -1,54 +1,148 @@
 import AnimateNumber from '@bankify/react-native-animate-number';
 import PropTypes from 'prop-types';
-import React, { useCallback } from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { Transition, Transitioning } from 'react-native-reanimated';
 import { colors } from '../../../styles';
 import { ColumnWithMargins, RowWithMargins } from '../../layout';
 import Spinner from '../../Spinner';
 import { Text } from '../../text';
 
+const showTimeRemainingDelay = 1000;
+
+const timeRemainingTransition = (
+  <Transition.Sequence>
+    <Transition.Together>
+      <Transition.Out durationMs={200} interpolation="easeOut" type="fade" />
+      <Transition.Out
+        delayMs={100}
+        durationMs={200}
+        interpolation="easeOut"
+        type="slide-bottom"
+      />
+    </Transition.Together>
+    <Transition.Change durationMs={100} interpolation="easeOut" />
+    <Transition.Together>
+      <Transition.In durationMs={200} interpolation="easeOut" type="fade" />
+      <Transition.In
+        delayMs={100}
+        durationMs={200}
+        interpolation="easeOut"
+        type="slide-bottom"
+      />
+    </Transition.Together>
+  </Transition.Sequence>
+);
+
+const unlockingTransition = (
+  <Transition.Sequence>
+    <Transition.Out durationMs={200} interpolation="easeInOut" type="fade" />
+    <Transition.Change durationMs={200} interpolation="easeInOut" />
+    <Transition.Together>
+      <Transition.In durationMs={100} interpolation="easeOut" type="fade" />
+      <Transition.In durationMs={100} interpolation="easeOut" type="scale" />
+    </Transition.Together>
+  </Transition.Sequence>
+);
+
 const renderCountdownText = animatedTimeRemaining => (
   <Text
     color={colors.alpha(colors.white, 0.4)}
     lineHeight="tight"
     size="smedium"
+    style={{
+      fontVariant: ['tabular-nums'],
+    }}
     weight="medium"
   >
     {`~ ${animatedTimeRemaining}s Remaining`}
   </Text>
 );
 
-const UnlockingSpinner = ({ interval, timeRemaining }) => {
+const UnlockingSpinner = ({
+  creationTimestamp,
+  estimatedApprovalTimeInMs,
+  interval,
+}) => {
+  const timeRemaining = useMemo(
```

Comment:
So in this case, timeRemaining is just a constant that we calculate based on the values of the creationTimestamp and estimatedApprovalTimeInMs (which don't change). We don't need this recalculated on the fly - given the dependency list, it will just use Date.now() at the time of change of the dependencies, which is basically just at the beginning. The "actual" dynamic remaining time will be calculated in the animation via timeRemaining - animatedValue

Reply:
TIL!